lib/util.js

```diff
@@ -613,13 +613,18 @@ function formatValue(ctx, value, recurseTimes) {
     }
   }
 
-  ctx.seen.push(value);
+  // TODO(addaleax): Make `seen` a Set to avoid linear-time lookup.
+  if (ctx.seen.includes(value)) {
```

Comment:
Yes. Or maybe that depends on what you mean by “object”; at least it’s not a primitive.